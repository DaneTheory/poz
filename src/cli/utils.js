import _ from '../logger/logger'

/**
 * Flattern cached packagesMap, used to log
 * @param packages {Array[POZPackage]}
 * @returns {Array}
 */
export function getPackagesMapLogData(packagesMap) {
  let packagesMapLogTableData = []
  Object.keys(packagesMap).forEach(packageName => {
    const pozPackage = packagesMap[packageName]
    // TODO remove it since we didn't have 'hide' package
    if (pozPackage.hide) {
      return;
    }
    let packagesMapLogTableDataRow = []
    packagesMapLogTableDataRow.push(packageName)
    packagesMapLogTableDataRow.push(pozPackage.origin)
    packagesMapLogTableDataRow.push(pozPackage.cachePath)
    packagesMapLogTableData.push(packagesMapLogTableDataRow)
  })
  return packagesMapLogTableData
}

/**
 * Log local packages
 * @param pkgMap
 */
export function localPackagesLogger(packagesMap) {
  let packagesMapLogTableData = getPackagesMapLogData(packagesMap)
  if (!packagesMapLogTableData.length) {
    return
  }
  _.echo('  ' + _.bold('local packages'.toUpperCase()))
  _.wrap()
  _.table(packagesMapLogTableData)
  _.wrap()
}

/**
 * errorList logger
 * @param errorList
 */
export function errorListLogger(packageName, errorList) {
  _.error(`package ${_.packageNameStyle(packageName)} is not a valid POZ package, see the error message below:`)
  for (let error of errorList) {
    _.wrap()
    _.echo(_.boldRed('*') + ' ' + error.message)
  }
  _.wrap()
  _.info(`You can use ${_.packageNameStyle('poz package -delete=' + packageName)} to remove this package`)
}

/**
 * Log local packages validate result
 */
export function localPackagesValidateResultLogger(packageValidationResultList) {
  for (let packageValidation of packageValidationResultList) {
    let { packageName, errorList } = packageValidation
    if (errorList.length) {
      errorListLogger(packageName, errorList)
    } else {
      _.success(`package ${_.packageNameStyle(packageName)} is a valid POZ package.`)
    }
  }
}

/**
 * Curry use function for CAC.
 * @param {function(cac)} commandFn
 * @returns {cac}
 */
export function curryUse(CLI) {
  return function use(commandFn) {
    const { command, options } = commandFn(CLI)
    const oldCommandHandler = command.handler
    const newCommandHandler = (input, flags) => {
      for (let i = 0, l = options.length; i < l; i++) {
        let option = options[i]
        if (flags[option.name]) {
          option.handler(input, flags)
          return
        }
      }
      oldCommandHandler(input, flags)
    }
    const cacCommand = CLI.command(command.name, command.opts, newCommandHandler)
    options.forEach(option => cacCommand.option(option.name, option.opts))
    return CLI
  }
}
