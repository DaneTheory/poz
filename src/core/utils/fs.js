import fs from 'fs'

export function exists(path) {
  return fs.existsSync(path)
}

export function isFile(path) {
  if (!exists(path)) {
    return null;
  }
  const stat = fs.statSync(path)
  return stat.isFile()
}

export function isDirectory(path) {
  if (!exists(path)) {
    return null;
  }
  const stat = fs.statSync(path)
  return stat.isDirectory()
}
