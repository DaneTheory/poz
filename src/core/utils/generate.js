


export function generate(sourceDir, targetDir, ) {

}
