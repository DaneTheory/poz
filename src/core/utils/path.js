import path from 'path'

export function resolve(...args) {
  return path.resolve(...args)
}
