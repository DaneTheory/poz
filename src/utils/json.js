export function beautifyJSON(ob) {
  return JSON.stringify(ob, null, 2)
}
