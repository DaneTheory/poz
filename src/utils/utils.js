const _toString = Object.prototype.toString

/**
 * Strict object type check. Only returns true
 * for plain JavaScript objects.
 */
export function isObject(obj) {
  return typeof str === 'object'
}

/**
 * Strict object type check. Only returns true
 * for plain JavaScript objects.
 */
export function isPlainObject(obj) {
  return _toString.call(obj) === '[object Object]'
}

/**
 * Array type check
 */
export function isArray(arr) {
  return Array.isArray(arr)
}

/**
 * Array type check
 */
export function isString(str) {
  return typeof str === 'string'
}

/**
 * Undefined type check
 */
export function isUndefined(str) {
  return typeof str === 'undefined'
}
