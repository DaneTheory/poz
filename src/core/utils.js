import { isString, isPlainObject, isFunction, isUndefined } from '../utils/datatypes'

export function assert(condition, msg) {
  if (!condition) throw new Error(`[poz] ${msg}`)
}

export function consolelog(condition, msg) {
  if (!condition) console.log(msg)
}

export function mergePOZDestConfig(dest, override = {}) {

  if (isString(override)) {
    dest.target = override
    return;
  }

  if (isFunction(override)) {
    override = override()
  } else if (!isPlainObject(override)) {
    throw new Error('Expect "dest" to be a string, function or a plain object')
  }

  Object.keys(dest).forEach(key => {
    if (!isUndefined(override[key])) {
      dest[key] = override[key]
    }
  });

  if (!isFunction(dest.render)) {
    throw new Error('Expect "dest.render" to be a function')
  }
}
