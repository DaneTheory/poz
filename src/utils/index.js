import { mixin } from './mixin'
import { exec, execSync } from './child_process'
import { getGitUser } from './git'
import { match } from './minimatch'
import { npmInstall, yarnInstall } from './npm'
import { simpleStringParser } from './parser'
import { getPkg } from './pkg'
import { promisify } from './promise'
import { assert, consolelog } from './assert'

import * as string from './string'
import * as prompts from './prompts'
import * as datatypes from './datatypes'

import fs from './fs'
import logger from './logger'

export {
  mixin,
  exec,
  execSync,
  getGitUser,
  match,
  npmInstall,
  yarnInstall,
  simpleStringParser,
  getPkg,
  promisify,
  assert,
  consolelog,
  string,
  prompts,
  datatypes,
  fs,
  logger
}

export default {
  mixin,
  exec,
  execSync,
  getGitUser,
  match,
  npmInstall,
  yarnInstall,
  simpleStringParser,
  getPkg,
  promisify,
  assert,
  consolelog,
  string,
  prompts,
  datatypes,
  fs,
  logger
}

