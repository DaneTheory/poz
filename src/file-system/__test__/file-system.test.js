import filex from '../'
import { beautifyJSON } from '../../utils/json'

describe('file-system', () => {

  const SRC_DIR = __dirname + '/fixtures/src'
  const DIST_DIR = __dirname + '/fixtures/dist'
  const node = filex(SRC_DIR)

  test('toJSON', () => {
    return node.traverse().then(() => {
      expect(node.toJSON()).toMatchSnapshot()
    })
  })

  test('flatten', () => {
    expect(node.flatten().map(li => li.basename)).toMatchSnapshot()
  })

  test('ignore', () => {
    node.ignore('d.js')
    console.log(beautifyJSON(node))
  })

  test('rename', () => {
    console.log(node)
    node.nodes[0].rename('NEW')
    console.log(beautifyJSON(node))
  })

  test('dest', () => {
    node.dest(DIST_DIR)
  })

})
