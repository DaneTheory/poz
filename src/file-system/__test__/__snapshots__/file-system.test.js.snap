// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`file-system flatten 1`] = `
Array [
  "a",
  "b",
  "c.js",
  "b.js",
  "d.js",
]
`;

exports[`file-system toJSON 1`] = `
Object {
  "basename": "src",
  "cwd": "/Users/haolchen/Documents/__self__/poz/src/file-system/__test__/fixtures/src",
  "nodes": Array [
    Object {
      "basename": "a",
      "cwd": "/Users/haolchen/Documents/__self__/poz/src/file-system/__test__/fixtures/src",
      "nodes": Array [
        Object {
          "basename": "b",
          "cwd": "/Users/haolchen/Documents/__self__/poz/src/file-system/__test__/fixtures/src",
          "nodes": Array [
            Object {
              "basename": "c.js",
              "cwd": "/Users/haolchen/Documents/__self__/poz/src/file-system/__test__/fixtures/src",
              "relative": "a/b/c.js",
            },
          ],
          "relative": "a/b",
        },
        Object {
          "basename": "b.js",
          "cwd": "/Users/haolchen/Documents/__self__/poz/src/file-system/__test__/fixtures/src",
          "relative": "a/b.js",
        },
      ],
      "relative": "a",
    },
    Object {
      "basename": "d.js",
      "cwd": "/Users/haolchen/Documents/__self__/poz/src/file-system/__test__/fixtures/src",
      "relative": "d.js",
    },
  ],
  "relative": "",
}
`;
